test/parser.test.ts

```diff
@@ -103,21 +103,21 @@ test('parser handles complex expressions', () => {
 
 test('parser throws on invalid input', () => {
   assert.throws(
-    () => new JQParser('@invalid').parse(),
+    () => new FGHParser('@invalid').parse(),
```

Comment:
As a developer I'd like to see error details, along witht the error position, for example:

`FGHParser('[')` => `unclosed token [`
`FGHParser('[0] something else')` => `unexpected expression "something else"`

Reply:
This is the kind of error you'd be getting:

```
ParseError: Parse error at position -1: Expected number, minus, or colon after [, got EOF
```

Will update the assertions